traits/ctraits.pyx

```diff
@@ -152,10 +152,13 @@ cdef object set_readonly_error(obj, name):
         "object.".format(name, type(obj))
     )
 
-cdef object invalid_attribute_error():
-    """ Raise an "attribute is not a string" error. """
 
-    raise TypeError('Attribute name must be a string.')
+cdef object invalid_attribute_error(name):
+    """ Raise an "attribute is not a string" error. """
+    raise TypeError(
+        (u"attribute name must be an instance of <type 'str'>. "
+         u"Got {0:!r} ({1:.200}).").format(name, str(type(name)))
```

Comment:
"{1!s:.200}" would save you the `str` call. I'd also suggest adding a limit on the length for the name; this wasn't there in the C code purely because the `%R` format specifier only started supporting a precision recently.